frontend/src/components/assessment-creation/DisplayAssessmentsPage.tsx

```diff
@@ -0,0 +1,287 @@
+  const [search, setSearch] = React.useState("");
+  const [sortProperty, setSortProperty] = React.useState("name");
+  const [sortOrder, setSortOrder] = React.useState("Ascending");
+
+  // const { loading, error, data } = useQuery(GET_USERS_BY_ROLE, {
+  //   fetchPolicy: "cache-and-network",
+  //   variables: { role: "Admin" },
+  // });
+
+  const filteredAssessements = React.useMemo(() => {
+    let filteredTests = data3;
+    if (search) {
+      filteredTests = filteredTests.filter(
+        (assessment: AssessmentType) =>
+          assessment.name.toLowerCase().includes(search.toLowerCase()) ||
+          assessment.country.toLowerCase().includes(search.toLowerCase()) ||
+          assessment.region.toLowerCase().includes(search.toLowerCase()) ||
+          assessment.status.toLowerCase().includes(search.toLowerCase()) ||
+          assessment.type.toLowerCase().includes(search.toLowerCase()),
+      );
+    }
+    return filteredTests?.map(getAssessments);
+  }, [search]);
+
+  const assessments = React.useMemo(() => {
+    let sortedAssessments: AssessmentType[] = filteredAssessements as AssessmentType[];
+    if (sortOrder === "descending") {
+      sortedAssessments = sortedAssessments?.sort((a, b) =>
+        a[sortProperty as keyof AssessmentType].toLowerCase() <
+        b[sortProperty as keyof AssessmentType].toLowerCase()
+          ? 1
+          : -1,
+      );
+    } else if (sortOrder === "ascending") {
+      sortedAssessments = sortedAssessments?.sort((a, b) =>
+        a[sortProperty as keyof AssessmentType].toLowerCase() >
+        b[sortProperty as keyof AssessmentType].toLowerCase()
+          ? 1
+          : -1,
+      );
+    }
+    return sortedAssessments;
+  }, [filteredAssessements, sortProperty, sortOrder]);
+
+  return (
+    <>
+      <Box>
+        <HStack justifyContent="space-between">
+          <Text
+            textStyle="header4"
+            color="blue.300"
+            style={{ textAlign: "left" }}
+            marginBottom="0.5em"
+          >
+            Assessments
+          </Text>
+          <CreateAssessementModel />
+        </HStack>
+      </Box>
+      {/* {loading && (
+        <Center margin="15%" flex="1">
+          <LoadingState />
+        </Center>
+      )}
+      {error && (
+        <Center margin="15%" flex="1">
+          <ErrorState />
+        </Center>
+      )} */}
+      {/* {data3 && !error && !loading && ( */}
+      <Box flex="1">
+        <Tabs marginTop={3}>
+          <TabList>
+            <Tab onClick={() => setSearch("")} color={unselectedColor}>
+              All
+            </Tab>
+            <Tab onClick={() => setSearch("Draft")} color={unselectedColor}>
+              Drafts
+            </Tab>
+            <Tab onClick={() => setSearch("Published")} color={unselectedColor}>
+              Published
+            </Tab>
+            <Tab onClick={() => setSearch("Archived")} color={unselectedColor}>
+              Archived
+            </Tab>
```

Comment:
I'm not sure if we should render table data for different tabs by using `setSearch` here. Since if someone subsequently searches in these tabs, it gets rid of the tab filter (i.e will not just show "Archived" assessments, will show all assessments that match the new search property).